articulate/sox
# Chore: update socket.io

# Description

Updates `socket.io-client` to the latest version. As this is an update to `4.x`, it is incompatible with `socket.io` server versions < `3.x` (see [socket.io compatibility table](https://socket.io/docs/v4/troubleshooting-connection-issues/#the-client-is-not-compatible-with-the-version-of-the-server)). Testing should be done with updated server versions in the PRs listed Links.

# Links

- [Issue](https://github.com/articulate/rise-authoring/issues/1970)
- [rise-frontend PR](https://github.com/articulate/rise-frontend/pull/4458)
- [conveyor PR](https://github.com/articulate/conveyor/pull/89)
- [Migrating from 2x to 3x](https://socket.io/docs/v4/migrating-from-2-x-to-3-0)
- [Migrating from 3x to 4x](https://socket.io/docs/v4/migrating-from-3-x-to-4-0)
- [Publishing an NPM package](https://www.notion.so/d7861c48d3154388a112760f423ecc0a?v=cab137f2ac6a4d6d8d4a9e23dff995eb&p=7694bc1a60504f109889c18ab2fffcd0&pm=s)

# Testing

<details>

<summary>Running locally</summary>

1. Run `yarn pack --filename sox-$(uuidgen).tgz`
2. Copy the tarball into the root directory of the target project (e.g. rise-frontend or rise-runtime)
3. Modify `Dockerfile` to copy tarball into docker container, example: https://github.com/articulate/rise-frontend/pull/2964/commits/1360ea4323cea4188d8e7efa4305f9f638972328#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R18 (lines 18 and 23)
4. Run `docker-compose run --rm app yarn add @articulate/sox@file:./sox-123.tgz`
  4a. Rename `sox-123.tgz` with name of tarball created in Step 1

</details>

For testing with rise-frontend, use: https://github.com/articulate/rise-frontend/pull/4458
For testing with conveyor, use https://github.com/articulate/conveyor/pull/89

- Ensure socket messages work as expected
